packages/react-instantsearch-core/src/components/__tests__/Index.js

```diff
@@ -157,4 +136,72 @@ describe.skip('Index', () => {
 
     expect(parameters.index).toBe('otherIndexName');
   });
+
+  it('wrapped with InstantSearchProvider: sets correct props', () => {
+    const state = {
+      results: {},
+      resultsFacetValues: {},
+      searching: false,
+      searchingForFacetValues: false,
+      isSearchStalled: false,
+      metadata: {},
+      error: {},
+      widgets: {
+        query: 'hello',
+      },
+    };
+
+    const context = {
+      onInternalStateUpdate() {},
+      createHrefForState() {},
+      onSearchForFacetValues() {},
+      onSearchStateChange() {},
+      onSearchParameters() {},
+      widgetsManager: {
+        registerWidget() {},
+        getWidgets() {},
+        update() {},
+      },
+      store: {
+        setState() {},
+        subscribe() {},
+        getState: () => state,
+      },
+    };
+
+    const Dummy = props => JSON.stringify(props, null, 2).replace(/"/g, '');
+
+    const Connected = createConnector({
+      displayName: 'Connector',
+      getProvidedProps: props => ({ providedProps: props }),
+    })(Dummy);
+
+    const props = {
+      message: 'hello',
+    };
+
+    const wrapper = mount(
+      <InstantSearchProvider value={context}>
+        <Index root={{ Root: 'div' }} indexId="test" indexName="test">
```

Comment:
We can use the `requiredProps` defined above. Here there is a confusion between `indexName` & `indexId` since both values are the same. The snapshots display the value but we don't know which one is it. We don't test this particular behaviour in the test but it brings a bit of confusion.